pkgs/native_assets_builder/test/build_runner/link_test.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test/test.dart';
+
+import '../helpers.dart';
+import 'helpers.dart';
+
+const Timeout longTimeout = Timeout(Duration(minutes: 5));
+
+void main() async {
+  test(
+    'simple_link linking',
+    timeout: longTimeout,
+    () async {
+      await inTempDir((tempUri) async {
+        await copyTestProjects(targetUri: tempUri);
+        final packageUri = tempUri.resolve('simple_link/');
+
+        // First, run `pub get`, we need pub to resolve our dependencies.
+        await runPubGet(
+          workingDirectory: packageUri,
+          logger: logger,
+        );
+
+        final buildResult = await build(
+          packageUri,
+          logger,
+          dartExecutable,
+        );
+        expect(buildResult.assets.length, 0);
+
+        final linkResult = await link(
+          packageUri,
+          logger,
+          dartExecutable,
+          buildResult: buildResult,
+        );
+        expect(linkResult.assets.length, 2);
+      });
+    },
+  );
+
+  test(
+    'complex_link linking',
+    timeout: longTimeout,
+    () async {
+      await inTempDir((tempUri) async {
+        await copyTestProjects(targetUri: tempUri);
+        final packageUri = tempUri.resolve('complex_link/');
+
+        // First, run `pub get`, we need pub to resolve our dependencies.
+        await runPubGet(
+          workingDirectory: packageUri,
+          logger: logger,
+        );
+
+        final buildResult = await build(
+          packageUri,
+          logger,
+          dartExecutable,
+        );
+        expect(buildResult.success, true);
+        expect(buildResult.assets.length, 2);
+
+        final linkResult = await link(
+          packageUri,
+          logger,
+          dartExecutable,
+          buildResult: buildResult,
+        );
+        expect(linkResult.success, true);
+        const assetsLeft = 2 + 2 - 1;
+        expect(linkResult.assets.length, assetsLeft);
```

Comment:
Add some expects of which assets are removed and are still there (based on assetId for example).